app/i18n/i18n.js

```diff
@@ -0,0 +1,95 @@
+var i18n = function() {
+  /**
+   * Internationalization util.
+   *
+   * @parm {string} locale The locale for message translations.
+   * @constructor
+   */
+  var I18nUtil = function (locale) {
+    var self = this;
+
+    /**
+     * Get locale
+     *
+     * @return {string} The current locale
+     */
+    this.getLocale = function () {
+      return self.locale;
+    };
+
+    /**
+     * Get translated message.
+     *
+     * Note: This function is used with the pf-i18n tag.
+     *
+     * @param {string} key The message key
+     */
+    this.getMsg = function (key) {
+      return self.jed.gettext(key);
+    };
+
+    /**
+     * Set locale
+     *
+     * @param {string} locale
+     */
+    this.setLocale = function (locale) {
+      self.locale = locale;
+      self._initLocaleData();
+    };
+
+    // Private functions
+
+    /**
+     * Fetch locale data
+     *
+     * @param url
+     * @private
+     */
+    this._fetchLocaleData = function(url) {
+      let xmlhttp = new XMLHttpRequest();
+      xmlhttp.onreadystatechange = function () {
+        if (this.readyState === 4 && this.status === 200) {
+          const localeData = JSON.parse(this.responseText);
+          self._setLocaleData(localeData);
+        }
+      };
+      xmlhttp.open("GET", url, false);
+      xmlhttp.send();
```

Comment:
I think a lot of frameworks are making use of the `fetch` polyfill/implementation now (see [here](https://github.com/github/fetch)), but happy to accept the more vanilla/cross supported use of `xmlHttp`. Great stuff.